src/aead/chacha20_poly1305.rs

```diff
@@ -278,6 +279,67 @@ mod tests {
                     (CRYPTO_POLY1305_STATE_LEN + 255) / 256);
     }
 
+    #[test]
+    pub fn chacha20_tests(){
+        test::from_file("src/chacha_tests.txt", |section, test_case| {
+            assert_eq!(section, "");
+
+            let key_bytes = test_case.consume_bytes("Key");
+            let mut key = [0u32; super::CHACHA20_KEY_LEN/4];
+            for ki in 0..super::CHACHA20_KEY_LEN/4 {
+                let kb = slice_as_array_ref!(&key_bytes[ki * 4..][..4], 4).unwrap();
+                key[ki] = polyfill::slice::u32_from_be_u8(kb);
+            }
+
+            let ctr_bytes = test_case.consume_bytes("Ctr");
+            let mut ctr = [0u32; 4];
+            for ci in 0..4 {
+                let cb = slice_as_array_ref!(&ctr_bytes[ci * 4..][..4], 4).unwrap();
+                ctr[ci] = polyfill::slice::u32_from_be_u8(cb);
+            }
+
+            let input = test_case.consume_bytes("Input");
+            let output = test_case.consume_bytes("Output");
+
+            for len in 0..input.len()+1 {
+                chacha20_test_case_inner(&key, &ctr, &input[..len], &output[..len], len);
+            }
+            Ok(())
+        });
+    }
+
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64",
+              target_arch = "arm", target_arch="aarch64"))]
+    const MAX_OFFSET: usize = 0;
+
+    #[cfg(not(any(target_arch = "x86", target_arch = "x86_64",
+                  target_arch = "arm", target_arch="aarch64")))]
+    const MAX_OFFSET: usize = 259;
+
+    fn chacha20_test_case_inner(key: &[u32; super::CHACHA20_KEY_LEN / 4],
+                                ctr: &[u32; 4], input: &[u8], output: &[u8], len: usize){
+        let mut outbuf = vec![0u8; len];
+
+        // Check regular operation is successful
```

Comment:
Please make comments like this a full sentence and end them with a period. (If you're not a native English speaker, let me know and I'll fix any grammar/spelling mistakes I notice for you.)